src/compositemark/boxplot.ts

```diff
@@ -145,3 +140,81 @@ export function normalizeBoxPlot(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT>
   };
 }
 
+export function box2DOrient(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | BoxPlotDef>): Orient {
+  const {mark: mark, encoding: encoding, ...outerSpec} = spec;
+  const xEncChannel = encoding.x as FieldDef<Field>;
+  const yEncChannel = encoding.y as FieldDef<Field>;
+  let resultOrient: Orient;
+
+  if (isContinuous(encoding.x) && isContinuous(encoding.y)) {
+    const xEncChannel = encoding.x as FieldDef<Field>;
+    const yEncChannel = encoding.y as FieldDef<Field>;
+    if (xEncChannel.aggregate === BOXPLOT && yEncChannel.aggregate === BOXPLOT) {
+      throw new Error('Both x and y cannot have aggregate');
+    }
+  } else if (isDiscrete(encoding.x) && isDiscrete(encoding.y)) {
+    throw new Error('Both x and y cannot be discrete');
+  }
+
+  if (isDiscrete(encoding.x) && isContinuous(encoding.y)) {
+    resultOrient = 'vertical';
+  } else if (isDiscrete(encoding.y) && isContinuous(encoding.x)) {
+    resultOrient = 'horizontal';
+  } else {
+    if (isContinuous(encoding.x) && isContinuous(encoding.y)) {
+      if (xEncChannel.aggregate === undefined && yEncChannel.aggregate === BOXPLOT) {
+        resultOrient = 'vertical';
+      } else if (yEncChannel.aggregate === undefined && xEncChannel.aggregate === BOXPLOT) {
+        resultOrient = 'horizontal';
+      } else if (xEncChannel.aggregate === BOXPLOT && yEncChannel.aggregate === BOXPLOT) {
+        return undefined; // invalid spec
+      } else {
+        const markChannel = mark as BoxPlotDef;
```

Comment:
Use type guard so you don't have to declare a new name (`markChannel` is a weird name anyway.)